### ISSUE TITLE
Many missing submodules to compile TangoSourceDistribution

### ISSUE THREAD

**Issue by sblanchet** [tango-controls/TangoSourceDistribution#6]
[`TangoSourceDistribution`](https://github.com/tango-controls/TangoSourceDistribution) needs many submodules to compile (for example  [`tango-idl`](https://github.com/tango-controls/tango-idl) or [`cppTango`](https://github.com/tango-controls/cppTango) )
It would be nice to have a `.gitmodules` file to specify all the required submodules, so that users can call
 `git clone --recursive` to have a ready-to-compile source directory in one step.

For comparison [`libhdbpp-mysql`](https://github.com/tango-controls/libhdbpp-mysql) has already such a [`.gitmodules`](https://github.com/tango-controls/libhdbpp-mysql/blob/master/.gitmodules) file to link to [`libhdbpp`](https://github.com/tango-controls/libhdbpp).

Actual error messages about missing submodules
```
me@tango-test:~$ git clone -q --recursive https://github.com/tango-controls/TangoSourceDistribution.git
me@tango-test:~$ cd TangoSourceDistribution/
me@tango-test:~/TangoSourceDistribution$ ./bootstrap
running aclocal -I m4
running libtoolize --copy --force
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'config'.
libtoolize: copying file 'config/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
running autoheader
running automake --gnu --add-missing
configure.ac:72: installing 'config/compile'
configure.ac:40: installing 'config/missing'
configure.ac:459: error: required file 'lib/idl/Makefile.in' not found
configure.ac:459: error: required file 'lib/cpp/server/Makefile.in' not found
configure.ac:459: error: required file 'lib/cpp/server/idl/Makefile.in' not found
configure.ac:459: error: required file 'lib/cpp/server/jpeg/Makefile.in' not found
configure.ac:459: error: required file 'lib/cpp/server/jpeg_mmx/Makefile.in' not found
configure.ac:459: error: required file 'lib/cpp/client/Makefile.in' not found
configure.ac:459: error: required file 'lib/cpp/client/helpers/Makefile.in' not found
configure.ac:459: error: required file 'cppserver/database/Makefile.in' not found
configure.ac:459: error: required file 'cppserver/database/create_db.sql.in' not found
configure.ac:459: error: required file 'cppserver/database/create_db.sh.in' not found
configure.ac:459: error: required file 'cppserver/database/my.cnf.in' not found
configure.ac:459: error: required file 'cppserver/database/stored_proc.sql.in' not found
configure.ac:459: error: required file 'cppserver/database/create_db_tables.sql.in' not found
configure.ac:459: error: required file 'cppserver/database/update_db.sh.in' not found
configure.ac:459: error: required file 'cppserver/database/update_db.sql.in' not found
configure.ac:459: error: required file 'cppserver/database/update_db8.sql.in' not found
configure.ac:459: error: required file 'cppserver/database/update_db7.sql.in' not found
configure.ac:459: error: required file 'cppserver/database/rem_history.sql.in' not found
configure.ac:459: error: required file 'cppserver/starter/Makefile.in' not found
configure.ac:459: error: required file 'cppserver/tangotest/Makefile.in' not found
configure.ac:459: error: required file 'cppserver/AbstractClass/AccessControl/Makefile.in' not found
configure.ac:459: error: required file 'cppserver/tangoaccesscontrol/Makefile.in' not found
configure.ac:459: error: required file 'utils/tango_admin/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/ds_writing/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/ds_model/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/dance/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/java_api/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/java_api/picture/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/gen_api/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/advanced/Makefile.in' not found
configure.ac:459: error: required file 'pogo/templates/Makefile.in' not found
configure.ac:459: error: required file 'pogo/templates/cpp/Makefile.in' not found
cppserver/AbstractClass/Makefile.am:1: error: required directory cppserver/AbstractClass/AccessControl does not exist
cppserver/Makefile.am:2: error: required directory cppserver/starter does not exist
cppserver/Makefile.am:2: error: required directory cppserver/tangotest does not exist
cppserver/Makefile.am:5: error: required directory cppserver/database does not exist
cppserver/Makefile.am:5: error: required directory cppserver/tangoaccesscontrol does not exist
doc/Makefile.am:3: error: required directory doc/src does not exist
lib/Makefile.am:2: error: required directory lib/idl does not exist
lib/cpp/Makefile.am:2: error: required directory lib/cpp/log4tango does not exist
lib/cpp/Makefile.am:2: error: required directory lib/cpp/server does not exist
lib/cpp/Makefile.am:2: error: required directory lib/cpp/client does not exist
pogo/Makefile.am:2: error: required directory pogo/templates does not exist
utils/Makefile.am:2: error: required directory utils/tango_admin does not exist
running autoconf
me@tango-test:~/TangoSourceDistribution$
```

**Comment by bourtemb** [comment]
Hi Sébastien.

Thank you very much for your help.
Using git submodules seems like the most natural/fastest way for this specific git repository, as it is done in HDB++. Even though in HDB++ we may end up using something different than sub-modules because it is not always easy to develop with them.
The latest Tango distribution was built from Sourceforge/subversion.
The next release of the Tango distribution will be built from GitHub/git.

As you noticed, there is still some work to be done to make it work as it used to work with Sourceforge/subversion.
We would like to use CMake to build the distribution too instead of using the auto-tools if possible. So this is quite some work.

This work will be done in the coming months in preparation of the new release but contributors are warmly welcomed of course.

If you want to install the latest Tango distribution from source (9.2.5a), you should do it from the tar file listed on the following link:
http://www.tango-controls.org/downloads/source/

Kind regards,
Reynald

**Comment by Ingvord** [comment]
I think this may be closed due to #8
